session_0/word_processing.rb

```diff
@@ -34,11 +49,35 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  score = 0
+  chars_1 = word_1.split('')
+  chars_2 = word_2.split('')
+  len_1 = chars_1.length - 1
+
+  for i in 0..len_1
+    if chars_1[i] == chars_2[i]
+      score += 1
+    else
+      break
+    end
+  end
+  return score
 end
 
 # Given a chosen word and an array of words, return an array of word(s)
 # with the maximum similarity score in the order they appear.
 def most_similar_words(chosen_word, words)
-  raise NotImplementedError # TODO
+  most_similar = []
+  max = 0
+  for word in words
```

Comment:
Prefer `each` over `for` loops: https://rubystyle.guide/#no-for-loops

Reply:
Ohk :+1: 
words.each do |word|
// Rest of code